plagiarism/process.go

```diff
@@ -1,75 +1,113 @@
 package plagiarism
 
 import (
-    "io"
-    "io/ioutil"
     "encoding/json"
-    "log"
-    "github.com/SpaceWhite/gomoss"
+    "github.com/inconshreveable/log15"
+    "github.com/omegaup/gomoss"
+    "github.com/omegaup/gluon/common"
+    "net/http"
+    "strconv"
     "fmt"
     "os"
 )
 
-// GUIDsStruct use for decoding JSON
-type GUIDsStruct struct {
+type Request struct {
+    ContestID uint32
     GUIDs []string
 }
 
-// Analysis struct contains output dir and reference dir
-type Analysis struct {
-    GIUDs []string
-    SubmissionDir string
-    Mossid uint32
-    Logger *log.Logger
-    ResultDir string
+type Response struct {
+    status int
 }
 
-// DecodeRequest decode request from request body
-func DecodeRequest(body io.ReadCloser) ([]string, error) {
-    decoder := json.NewDecoder(body)
-    var guidsStruct GUIDsStruct
-    err := decoder.Decode(&guidsStruct)
+type AnalyzerContext struct {
+    SubmissionPath string
+    ResultPath string
+    ID common.IDConfig
+}
+
+type Analyzer struct {
+    ctx AnalyzerContext
+    log log15.Logger
+    queue chan Request
+    done chan bool
+}
 
-    if err != nil {
-        return []string{}, err
+func NewAnalyzer(ctx common.Context) (*Analyzer, error){
+    analyzer := Analyzer{
+        ctx: AnalyzerContext{
+            SubmissionPath: ctx.Config.Plagiarism.Path.Submission,
+            ResultPath: ctx.Config.Plagiarism.Path.Result,
+            ID: ctx.Config.Plagiarism.ID,
+        },
+        log: ctx.Log,
     }
-    return guidsStruct.GUIDs, nil
+    return &analyzer, nil
 }
 
 // GUIDPath use to get submission from OmegaUp
 // Example GUID : abcdefghi1234
 // GUID Path    : submissionDir/ab/cdefghi1234
-func (analysis *Analysis) GUIDPath(guid string) string {
-    return fmt.Sprintf("%v/%v/%v", analysis.SubmissionDir, guid[:2], guid[2:])
+func (analyzer *Analyzer) GUIDPath(guid string) string {
+    return fmt.Sprintf("%v/%v/%v", analyzer.ctx.SubmissionPath, guid[:2], guid[2:])
```

Comment:
Can you use [`path.Join`](https://golang.org/pkg/path/#Join) instead? Same everywhere you concatenate subpaths.